fidesctl/.fides/dataset.yml

```diff
@@ -6,6 +6,7 @@ dataset:
     meta: {}
     data_categories:
       - system.operations
+    retention: "1 year post employment"
```

Comment:
This will add a null value across every field within both `annotate-dataset` and `generate-dataset` (the same way nested fields do currently) - is this behavior we want to eventually provide an option for as well to remove the null items?

Reply:
Planning to add a follow on issue for these to not hold up working through the remaining model items

Reply:
for generate dataset we can leave it blank, but i'd expect annotate dataset to surface that null value

Reply:
Not sure if this is what you had in mind but I updated `generate-dataset` to exclude both optional retention and nested fields if empty, @earmenda had some thoughts around populating these as well that we [discussed in this PR thread](https://github.com/ethyca/fides/pull/316#issuecomment-1016065574)

Reply:
I'm kind of torn on this like @earmenda is...

if we leave it null in the generated dataset, then it is really easy for `annotate dataset` to catch that and prompt the user to fill it in, which is maybe the behavior we want? 

If we imagine someone generating a system, and then running `annotate system` we would definitely want to prompt them to add things like `retention`, and doing that without having them already there as null fields would make things trickier but not impossible

Reply:
maybe we add a flag to `generate` that lets the user choose whether to keep null values or not?

Reply:
From the nested fields perspective, I think it makes sense to exclude them every time unless required and populate them as required once we have a connector for Mongo to test with in here, otherwise it's likely just noise in the yml

Retention I can see some value being added, at least at the top level but maybe not for each and every field... I can  add the flag and default it to exclude the nulls (for now) and then also see about adding a default value to the dataset/collection level for retention so those conspicuously populate at a high level - would that make sense to you?

Reply:
I think that we can control the `annotate-dataset` command separately with a flag to populate the nulls in the end as well (if I understand things correctly) by not having `exclude_none=True` on the output - maybe a flag or command prompt there as well

Reply:
Made some changes to address this, adding flags like suggested 👍🏽 

I am considering moving work to update `annotate-dataset` to handle more than just `data_categories` to a separate issue to give some time to think about the user experience of that and keep moving with the other attributes